collector/textfile.go

```diff
@@ -271,10 +301,20 @@ fileLoop:
 		// a failure does not appear fresh.
 		mtimes[f.Name()] = f.ModTime()
 
-		for _, mf := range parsedFamilies {
+		for _, metricFamily := range parsedFamilies {
+			metricFamilies = append(metricFamilies, metricFamily)
+		}
+	}
+
+	if duplicateMetricEntry(metricFamilies) {
+		log.Errorf("Duplicate metrics detected in files")
+		error += 1.0
+	} else {
+		for _, mf := range metricFamilies {
 			convertMetricFamily(mf, ch)
 		}
 	}
+	// If crash is caused by convertMetricFamily above, move to an else {} block here
 
 	c.exportMTimes(mtimes, ch)
```

Comment:
Should we still export the mtimes if there are duplicates? I'd expect us to only emit the scrape_error metric in that case?

Reply:
Another good spot; I've moved this so it's only run if no duplicates are detected.